Sources/Support/Networking/HTTPMethod.swift

```diff
@@ -1,46 +1,39 @@
 import Foundation
 
 /// An HTTP method.
-public enum HTTPMethod: String, Equatable {
-    case get = "GET"
-    case post = "POST"
-    case put = "PUT"
-    case delete = "DELETE"
-    case options = "OPTIONS"
-    case connect = "CONNECT"
-    case head = "HEAD"
-    case patch = "PATCH"
-    case trace = "TRACE"
+public enum HTTPMethod: Equatable{
+    case get
+    case post(body: HTTPRequest.Body)
+    case put(body: HTTPRequest.Body)
+    case delete(body: HTTPRequest.Body?)
```

Comment:
I read at [developer.mozzila](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE) that DELETE method _may_ include body. After your comment, I went to the official documentation and indeed it's allowed to have body in the DELETE request but we shouldn't include it... I guess we should focus on **"SHOULD NOT"**. 😄 

[RFC 9110: HTTP Semantics](https://www.rfc-editor.org/rfc/rfc9110#name-delete)
>[...] content received in a DELETE request has no generally defined semantics, cannot alter the meaning or target of the request, and might lead some implementations to reject the request and close the connection [...]. A client **SHOULD NOT** generate content in a DELETE request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported. An origin server **SHOULD NOT** rely on private agreements to receive content, since participants in HTTP communication are often unaware of intermediaries along the request chain.
This language has been strengthened from the previous language, to say that even though it is allowed, you really need to be very careful when using it because (for example) some users might be behind a proxy that would strip the body from the request in order to combat ["request smuggling"](https://www.rfc-editor.org/rfc/rfc9112#section-11.2).

Reply:
Thanks. Yes, the SHOULD NOT is exactly the wording used also for e.g. get, so let’s rely on it.

I made a separate PR to update docs to the new standard if you’d like to have a look. #16